dialog_design.md

```diff
@@ -0,0 +1,68 @@
+# Dialogs
+
+Goal: Display a message and an optional graphic to all users, or some, or one, and return a response.
+
+## Architecture
+
+Create a new `Dialog` class that manages calls and responses. It is also a good place to park static convenience
+functions and event listeners. I propose a couple different setup options:
+
+1. `type` - What sort of response we should expect.
+	1. A `message` dialog presents an OK button, and returns no value, just a confirmation that OK was clicked.
+	2. A `confirm` dialog presents OK and Cancel buttons, and returns a boolean.
+	3. A `prompt` dialog presents a text field and OK/Cancel buttons, and returns the contents of the text field.
+2. `text` - The text that the dialog displays.
+3. `recipients` (optional) - The dialog is displayed to all currently connected users by default, but can be restricted
+	to a subset of users with this option. Takes either a `GroupMask`, a `User[]`, or just one `User`. This filtering
+	could be implemented in the sync layer fairly easily, I think.
+4. `icon` (optional) - A graphic to display alongside the text. Can be one of the built-in icons (`info`, `warning`,
+	`error`, or `question`), or a custom texture supplied by the developer.
+5. `persistent` (optional) - If `true`, the dialog is presented to users in the recipients list even if they are
```

Comment:
Let's defer `persistent` in v1. Sounds tricky, and there's enough value here without the complication.